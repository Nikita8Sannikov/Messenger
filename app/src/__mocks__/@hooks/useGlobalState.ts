const mockUseAppSelector = jest.fn();
const mockUseAppDispatch = jest.fn();

export const useAppSelector = mockUseAppSelector;
export const useAppDispatch = mockUseAppDispatch;

export { mockUseAppSelector, mockUseAppDispatch };
