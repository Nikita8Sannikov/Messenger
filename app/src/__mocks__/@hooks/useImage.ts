const useImage = (src: string | null): string => {
	return src || "/assets/images/noPhoto.jpg";
};

export default useImage; 