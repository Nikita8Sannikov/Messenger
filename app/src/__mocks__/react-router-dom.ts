import { createElement, type ReactNode } from "react";
import * as ReactRouterDom from "react-router-dom";

const mockNavigate = jest.fn();
const mockLocation = {
	pathname: "/",
	search: "",
	hash: "",
	state: null,
	key: "default",
};

const mockParams = {};

const useNavigate = () => mockNavigate;
const useLocation = () => mockLocation;
const useParams = () => mockParams;

const matchPath = (pattern: string, pathname: string) => {
	if (pattern === pathname) {
		return {
			params: {},
			pathname,
			pattern,
			pathnameBase: pathname,
		};
	}
	return null;
};

const Routes = ({ children }: { children: ReactNode }) => children;
const Route = ({ children }: { children: ReactNode }) => children;
const Navigate = ({ to }: { to: string }) => createElement("div", { "data-testid": "navigate", "data-to": to });

export { useNavigate, mockNavigate, useLocation, mockLocation, useParams, mockParams, matchPath, Routes, Route, Navigate };
export default {
	...ReactRouterDom,
	useNavigate,
	useLocation,
	useParams,
	matchPath,
	Routes,
	Route,
	Navigate,
}; 