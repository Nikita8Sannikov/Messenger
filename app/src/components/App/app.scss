@import "../../styles/variables.scss";
@import "../../styles/colors.scss";

.root {
    min-height: 100vh;
    width: 100%;
    background: $mainBackground;

    &__wrapper {
        width: inherit;
        display: flex;
        align-items: center;
        justify-content: center;
        padding-bottom: 20px;

        &__container {
            width: $contentWidth;
            display: flex;
            align-items: flex-start;
            justify-content: center;

            &__no-auth {
                width: 100%;
            }

            &__content {
                flex: 1;
                margin-top: $marginTop;
            }
        }
    }
}