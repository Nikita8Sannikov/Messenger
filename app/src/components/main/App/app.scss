.root {
	min-height: 100vh;
	width: 100%;
	background: $main-background;
	overflow-x: hidden;
	position: relative;

	&__wrapper {
		width: inherit;
		display: flex;
		align-items: center;
		justify-content: center;
		padding-bottom: 20px;

		&__container {
			width: $content-width;
			display: flex;
			align-items: flex-start;
			justify-content: center;

			&__no-auth {
				width: 100%;
			}

			&__content {
				flex: 1;
				margin-top: 20px;
			}
		}
	}
}
