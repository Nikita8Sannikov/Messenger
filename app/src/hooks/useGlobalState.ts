import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { AppDispatch, RootState } from "@custom-types/redux.types";

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector<RootState>;