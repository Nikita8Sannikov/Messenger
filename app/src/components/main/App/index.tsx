import Router from "@components/main/Router";
import ServiceComponents from "@components/main/ServiceComponents";
import HeaderComponent from "@components/layouts/header";
import MenuComponent from "@components/layouts/menu";
import SpinnerComponent from "@components/ui/spinner";
import { selectMainState } from "@store/main/slice";
import { useAppSelector } from "@hooks/useGlobalState";

import "./app.scss";

// Главный компонент, который отрисовывает основную верстку проекта
export default function App() {
	const { isAuth, loading } = useAppSelector(selectMainState);
	
	return <div className="root">
		<ServiceComponents />

		{loading
			? <SpinnerComponent />
			: <>
				{isAuth ? <HeaderComponent /> : null}

				<div className="root__wrapper">
					{/* <button onClick={() => setIsDarkMode((prev: boolean) => !prev)}>Toggle Theme</button> */}
					<div
						className={`root__wrapper__container ${
							isAuth
								? ""
								: "root__wrapper__container__no-auth"
						}`}
					>
						{isAuth ? <MenuComponent /> : null}

						<div className="root__wrapper__container__content">
							<Router />
						</div>
					</div>
				</div>
			</>
		}
	</div>
}
