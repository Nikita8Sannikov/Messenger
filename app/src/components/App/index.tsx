import React from "react";

import { selectMainState } from "../../state/main/slice";
import { useAppSelector } from "../../hooks/useGlobalState";
import Router from "../../service/Router";
import ServiceComponents from "../../service/ServiceComponents";
import Header from "../Header";
import MenuComponent from "../Menu";

import "./app.scss";

export default function App() {
  const { isAuth, loading } = useAppSelector(selectMainState);

  return <div className="root">
    <ServiceComponents />

    {loading
      ? null
      : <>
        {isAuth ? <Header /> : null}

        <div className="root__wrapper">
          <div className={`root__wrapper__container ${isAuth ? "" : "root__wrapper__container__no-auth"}`}>
            {isAuth ? <MenuComponent /> : null}

            <div className="root__wrapper__container__content">
              <Router />
            </div>
          </div>
        </div>
      </>
    }
  </div>
};