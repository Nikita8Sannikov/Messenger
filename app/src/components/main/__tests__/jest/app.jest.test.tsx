import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";

import App from "@components/main/app";
import { mockUseAppSelector } from "../../../../__mocks__/@hooks/useGlobalState";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

jest.mock("@components/services/modals/ModalWithAttachments", () => ({
  __esModule: true,
  default: () => <div data-testid="mocked-modal-attachments">Mocked Modal With Attachments</div>,
}));

jest.mock("@components/layouts/header", () => ({
  __esModule: true,
  default: () => <div data-testid="mocked-header">Mocked Header</div>,
}));

jest.mock("@components/layouts/menu", () => ({
  __esModule: true,
  default: () => <div data-testid="mocked-menu">Mocked Menu</div>,
}));

jest.mock("@components/main/ServiceComponents", () => ({
  __esModule: true,
  default: () => <div >Mocked ServiceComponents</div>,
}));

jest.mock("@components/main/Router", () => ({
  __esModule: true,
  default: () => <div >Mocked Router</div>,
}));

describe("AppComponent", () => {

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("all shows correct if loading: true, isAuth: false ", () => {
    mockUseAppSelector.mockReturnValue({ loading: true, isAuth: false })
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )

    const spinner = screen.getByTestId("spinner");
    const header = screen.queryByTestId("header");
    const menu = screen.queryByTestId("menu");
    expect(spinner).toBeInTheDocument();
    expect(header).not.toBeInTheDocument();
    expect(menu).not.toBeInTheDocument();
  })

  test("all shows correct if loading: false, isAuth: true ", () => {


    mockUseAppSelector.mockReturnValue({ loading: false, isAuth: true })
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )

    const spinner = screen.queryByTestId("spinner");
    const header = screen.queryByTestId("mocked-header");
    const menu = screen.queryByTestId("mocked-menu");
    expect(spinner).not.toBeInTheDocument();
    expect(header).toBeInTheDocument();
    expect(menu).toBeInTheDocument();
  })

  test("all shows correct if loading: false, isAuth: false ", () => {


    mockUseAppSelector.mockReturnValue({ loading: false, isAuth: false })
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )

    const spinner = screen.queryByTestId("spinner");
    const header = screen.queryByTestId("mocked-header");
    const menu = screen.queryByTestId("mocked-menu");
    expect(spinner).not.toBeInTheDocument();
    expect(header).not.toBeInTheDocument();
    expect(menu).not.toBeInTheDocument();
  })

  test("matches snapshot", () => {
    const { asFragment } = render(<App></App>);

    // Снимаем снапшот компонента
    expect(asFragment()).toMatchSnapshot();
  });
})