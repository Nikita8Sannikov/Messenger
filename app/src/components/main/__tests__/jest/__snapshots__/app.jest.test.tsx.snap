// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div>
      Mocked ServiceComponents
    </div>
    <div
      class="root__wrapper"
    >
      <div
        class="root__wrapper__container root__wrapper__container__no-auth"
      >
        <div
          class="content MuiBox-root css-0"
        >
          <div
            class="root__wrapper__container__content"
          >
            <div>
              Mocked Router
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
